src/pods/generate/TemplatedResourcesGenerator.ts

```diff
@@ -72,29 +86,41 @@ export class TemplatedResourcesGenerator implements ResourcesGenerator {
   }
 
   /**
-   * Generates ResourceLinks for each entry in the given folder.
+   * Creates a TemplateResourceLink for the given filePath.
+   * The identifier will be based on the file path stripped from the template extension,
+   * but the filePath parameter will still point to the original file.
    */
-  private async* generateLinks(folderPath: string, mapper: FileIdentifierMapper): AsyncIterable<ResourceLink> {
-    const files = await fsPromises.readdir(folderPath);
-    for (const name of files) {
-      const filePath = joinFilePath(folderPath, name);
-      const stats = await fsPromises.lstat(filePath);
-      yield mapper.mapFilePathToUrl(filePath, stats.isDirectory());
-    }
+  private async toTemplateLink(filePath: string, mapper: FileIdentifierMapper): Promise<TemplateResourceLink> {
+    const stats = await fsPromises.lstat(filePath);
+
+    // Slice the template extension from the filepath for correct identifier generation
+    const isTemplate = filePath.endsWith(this.templateExtension);
+    const slicedPath = isTemplate ? filePath.slice(0, -this.templateExtension.length) : filePath;
+    const link = await mapper.mapFilePathToUrl(slicedPath, stats.isDirectory());
+    // We still need the original file path for disk reading though
+    return {
+      ...link,
+      filePath,
+      isTemplate,
+      isMetadata: link.identifier.path.endsWith(this.metaExtension),
```

Comment:
Shouldn't that be determined by a strategy? This is also encoded in `FileDataAccessor`.

Reply:
True, we could have the `FileDataAccessor` expose this information. But on the other hand there is no real reason why they should have the same behaviour since one is visible to the user and one is not. But no strong opinion so I'll go with what you prefer.

Reply:
The TemplatedResourcesGenerator is going to disk, so de facto reusing some of the `FileDataAccessor` conventions.

If not too much overhead, let's see what happens if we do it. Better to not encode information twice.